src/main/scala/com/campudus/tableaux/cache/CacheVerticle.scala

```diff
@@ -0,0 +1,202 @@
+    }
+  }
+
+  private def messageHandlerRetrieve(message: Message[JsonObject]): Unit = {
+    val obj = message.body()
+
+    (for {
+      tableId <- Option(obj.getLong("tableId")).map(_.toLong)
+      columnId <- Option(obj.getLong("columnId")).map(_.toLong)
+      rowId <- Option(obj.getLong("rowId")).map(_.toLong)
+    } yield (tableId, columnId, rowId)) match {
+      case Some((tableId, columnId, rowId)) =>
+
+        implicit val scalaCache = getCache(tableId, columnId)
+
+        get[AnyRef, NoSerialization](rowId)
+          .map({
+            case Some(value) =>
+              val reply = Json.obj(
+                "tableId" -> tableId,
+                "columnId" -> columnId,
+                "rowId" -> rowId,
+                "value" -> value
+              )
+
+              message.reply(reply)
+            case None =>
+              logger.warn(s"messageHandlerRetrieve $tableId, $columnId, $rowId not found")
+              message.fail(NOT_FOUND_FAILURE, "Not found")
+          })
+
+      case None =>
+        logger.error("Message invalid: Fields (tableId, columnId, rowId) should be a Long")
+        message.fail(INVALID_MESSAGE, "Message invalid: Fields (tableId, columnId, rowId) should be a Long")
+    }
+  }
+
+  private def messageHandlerInvalidate(message: Message[JsonObject]): Unit = {
+    val obj = message.body()
+
+    (for {
+      tableId <- Option(obj.getLong("tableId")).map(_.toLong)
+      columnId <- Option(obj.getLong("columnId")).map(_.toLong)
+    } yield (tableId, columnId, Option(obj.getLong("rowId")).map(_.toLong))) match {
+      case Some((tableId, columnId, Some(rowId))) =>
+
+        implicit val scalaCache = getCache(tableId, columnId)
+
+        remove(rowId)
+          .map({
+            _ =>
+              val reply = Json.obj(
+                "tableId" -> tableId,
+                "columnId" -> columnId,
+                "rowId" -> rowId
+              )
+
+              message.reply(reply)
+          })
+
+      case Some((tableId, columnId, None)) =>
+
+        implicit val scalaCache = getCache(tableId, columnId)
+
+        removeAll()
+          .map({
+            _ =>
+              removeCache(tableId, columnId)
+
+              val reply = Json.obj(
+                "tableId" -> tableId,
+                "columnId" -> columnId
+              )
+
+              message.reply(reply)
+          })
+
+      case None =>
+        logger.error("Message invalid: Fields (tableId, columnId) should be a Long")
+        message.fail(INVALID_MESSAGE, "Message invalid: Fields (tableId, columnId) should be a Long")
+    }
+  }
+
+  private def messageHandlerInvalidateAll(message: Message[JsonObject]): Unit = {
+    Future.sequence(caches.map({
+      case ((tableId, columnId), cache) =>
+        removeAll()(cache)
+          .map({
+            _ =>
+              removeCache(tableId, columnId)
+          })
+    })).onComplete({
+      case _ =>
+        caches.clear()
```

Comment:
Does it make sense to remove everything from the cache, not looking for failures and then clearing the map anyways -> loosing all references to the cache instances?